panel/lxqtpanel.cpp

```diff
@@ -1407,48 +1416,60 @@ Plugin* LXQtPanel::findPlugin(const ILXQtPanelPlugin* iPlugin) const
  ************************************************/
 QRect LXQtPanel::calculatePopupWindowPos(QPoint const & absolutePos, QSize const & windowSize) const
 {
-    int x = absolutePos.x(), y = absolutePos.y();
+    QPoint localPos = mapFromGlobal(absolutePos);
+    int x = localPos.x(), y = localPos.y();
 
     switch (position())
     {
     case ILXQtPanel::PositionTop:
-        y = mGeometry.bottom();
+        y = geometry().height();
         break;
 
     case ILXQtPanel::PositionBottom:
-        y = mGeometry.top() - windowSize.height();
+        y = 0 - windowSize.height();
         break;
 
     case ILXQtPanel::PositionLeft:
-        x = mGeometry.right();
+        x = geometry().right();
         break;
 
     case ILXQtPanel::PositionRight:
-        x = mGeometry.left() - windowSize.width();
+        x = geometry().left() - windowSize.width();
         break;
     }
 
     QRect res(QPoint(x, y), windowSize);
 
-    QRect panelScreen;
-    const auto screens = QApplication::screens();
-    if (mActualScreenNum < screens.size())
-        panelScreen = screens.at(mActualScreenNum)->geometry();
-    // NOTE: We cannot use AvailableGeometry() which returns the work area here because when in a
-    // multihead setup with different resolutions. In this case, the size of the work area is limited
-    // by the smallest monitor and may be much smaller than the current screen and we will place the
-    // menu at the wrong place. This is very bad for UX. So let's use the full size of the screen.
-    if (res.right() > panelScreen.right())
-        res.moveRight(panelScreen.right());
+    // Map to global coordinates
+    res = QRect(mapToGlobal(res.topLeft()), mapToGlobal(res.bottomRight()));
```

Comment:
Supposing that the mentioned issues are fixed, this converting of global to local and then returning to the first coordinates seem redundant, whether under X11 or under Wayland. The only meaningful change that has happened is using of `geometry()` instead of `mGeometry`.